packages/flutter/lib/src/rendering/object.dart

```diff
@@ -2841,6 +2887,8 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
       header += ' NEEDS-COMPOSITING-BITS-UPDATE';
     if (!attached)
       header += ' DETACHED';
+    if (_debugDisposed)
```

Comment:
maybe move this higher up in the list and short circuit all the other checks if the object is disposed? NEEDS-LAYOUT, NEEDS-PAINT, etc are probably all misleading if the object has been disposed.